tests/apps/test_graph_utils.py

```diff
@@ -186,3 +186,91 @@ def test_invalid_subgraphs(self, graph, dim):
         subgraphs = itertools.combinations(range(dim), int(dim / 2))
 
         assert not all([utils.is_subgraph(list(s), graph) for s in subgraphs])
+
+
+@pytest.mark.parametrize("dim", range(2, 10))
+class TestIsClique:
+    """Tests for the function `strawberryfields.apps.graph.utils.is_clique` """
+
+    def test_no_false_negatives(self, dim):
+        """ Tests that cliques are labelled as such"""
+        g = nx.complete_graph(dim)
+        assert utils.is_clique(g)
+
+    def test_no_false_positives(self, dim):
+        """ Tests that non-cliques are labelled as such"""
+        g = nx.empty_graph(dim)
+        assert not utils.is_clique(g)
+
+
+@pytest.mark.parametrize("dim", range(2, 10))
+class TestC0:
+    """ Tests function c_0 that generates the set of nodes connected to all nodes in a clique"""
+
+    def test_grows_to_clique(self, dim):
+        """ Tests that adding nodes from C0 to a clique results in another clique"""
+        A = nx.complete_graph(dim)
+        S = [0]
+        K = utils.c_0(S, A)
+        while K:
+            new_node = K[0]
+            S.append(new_node)
+            K = utils.c_0(S, A)
+
+        assert utils.is_clique(A.subgraph(S))
+
+    def test_c0_comp_graph(self, dim):
+        """ Tests that the set C0 for a node in a clique consists of all remaining nodes"""
+        A = nx.complete_graph(dim)
+        S = [dim - 1]
+        K = utils.c_0(S, A)
+
+        assert K == list(range(dim - 1))
+
+    def test_c0input_is_clique(self, dim):
+        """ Tests if function raises a ``ValueError`` when input is not a clique"""
+        A = np.ones((dim, dim)) - np.eye(dim)
+        A = nx.Graph(A)
+        A.remove_edge(0, 1)
+        S = [0, 1]
+
+        with pytest.raises(ValueError, match="Input subgraph is not a clique"):
+            utils.c_0(S, A)
+
+
+@pytest.mark.parametrize("dim", range(4, 10))
+class TestC1:
+    """Tests function c_1 that generates the set of nodes connected to all *but one* of the nodes
+    in a clique"""
+
+    def test_c1_comp_graph(self, dim):
+        """ Tests that c1 set is correctly generated for an almost-complete graph, where edge
+        (0, 1) is removed """
+        A = nx.complete_graph(dim)
+        A.remove_edge(0, 1)
+        S = [i for i in range(1, dim)]
+        c1 = utils.c_1(S, A)
+
+        assert c1 == [(1, 0)]
+
+    def test_c1_swap_to_clique(self, dim):
+        """ Tests that c1 set gives a valid clique after swapping """
+        A = nx.complete_graph(dim)
+        A.remove_edge(0, 1)
+        S = [i for i in range(1, dim)]
+        c1 = utils.c_1(S, A)
+        swap_nodes = c1[0]
+        S.remove(swap_nodes[0])
+        S.append(swap_nodes[1])
+
+        assert utils.is_clique(A.subgraph(S))
+
+    def test_c1input_is_clique(self, dim):
```

Comment:
```suggestion
    def test_c1_input_is_clique(self, dim):
```

Reply:
I see this is resolved, but there's no change?

Reply:
It eventually got changed to
`test_c_1_error_on_not_clique_input`
which combines your suggestion (underscore) with @co9olguy's suggestion on clearer naming